src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBuilder.test.tsx

```diff
@@ -0,0 +1,321 @@
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import mapFullTableMeta from '@common/modules/table-tool/utils/mapFullTableMeta';
+import mapFullTable from '@common/modules/table-tool/utils/mapFullTable';
+import { TableDataSubjectMeta } from '@common/services/tableBuilderService';
+import { render, screen, waitFor, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import noop from 'lodash/noop';
+import React from 'react';
+import ChartBuilder from '../ChartBuilder';
+
+describe('ChartBuilder', () => {
+  const testFormState: ChartBuilderForms = {
+    options: {
+      isValid: true,
+      submitCount: 0,
+      id: 'options',
+      title: 'Options',
+    },
+    dataSets: {
+      isValid: true,
+      submitCount: 0,
+      id: 'dataSets',
+      title: 'Data sets',
+    },
+  };
+
+  const testTable = mapFullTable(testTableData);
+  const testMeta: TableDataSubjectMeta = {
+    publicationName: '',
+    subjectName: '',
+    geoJsonAvailable: false,
+    footnotes: [],
+    boundaryLevels: [],
+    locations: {
+      localAuthority: [
+        { id: 'barnet', label: 'Barnet', value: 'barnet' },
+        { id: 'barnsley', label: 'Barnsley', value: 'barnsley' },
+      ],
+    },
+    timePeriodRange: [
+      {
+        label: '2019/20',
+        year: 2019,
+        code: 'AY',
+      },
+      {
+        label: '2020/21',
+        year: 2020,
+        code: 'AY',
+      },
+    ],
+    indicators: [
+      {
+        value: 'authorised-absence-sessions',
+        label: 'Number of authorised absence sessions',
+        unit: '',
+        name: 'sess_authorised',
+        decimalPlaces: 2,
+      },
+      {
+        value: 'unauthorised-absence-sessions',
+        label: 'Number of unauthorised absence sessions',
+        unit: '',
+        name: 'sess_unauthorised',
+        decimalPlaces: 2,
+      },
+    ],
+    filters: {
+      Characteristic: {
+        legend: 'Characteristic',
+        name: 'characteristic',
+        options: {
+          gender: {
+            id: 'gender',
+            label: 'Gender',
+            options: [
+              {
+                value: 'male',
+                label: 'Male',
+              },
+              {
+                value: 'female',
+                label: 'Female',
+              },
+            ],
+            order: 0,
+          },
+        },
+        order: 0,
+      },
+    },
+  };
+  const testSubjectMeta: FullTableMeta = mapFullTableMeta(testMeta);
+
+  test('renders the chart type selector only when no initial value', () => {
+    render(
+      <ChartBuilderFormsContextProvider initialForms={testFormState}>
+        <ChartBuilder
+          releaseId="release-1"
+          data={testTable.results}
+          meta={testSubjectMeta}
+          tableTitle="Table title"
+          onChartSave={jest.fn(() => Promise.resolve())}
```

Comment:
Bit of a nitpick, but couldn't we just pass in `jest.fn()` for all `onChartSave` and `onTableQueryUpdate` props?